### PR TITLE
Assignment 3 Pull Request

### PR DESCRIPTION
**Pull request by UKJWL**

### CODE REVIEW

File: force-app/main/default/lwc/timeline/timeline.html [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+                    
+                        <template if:true={CompletedTasks}>
+
+                      
+                          <div class="task-card" >
+                            <ul class="slds-has-dividers_around-space"> 
+                              <li class="slds-item">
+                                
+                                <article class="slds-tile slds-tile_board">
+                                  <h3 class="slds-tile__title slds-truncate" title="Journey Name">
```
Comment (Drywkenway) [inline force-app/main/default/lwc/timeline/timeline.html:25]: why is the title journey name? im assuming you meant task name.

File: force-app/main/default/lwc/timeline/timeline.html [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+                    
+                        <template if:true={InProgressTasks}>
+
+                      
+                          <div class="task-card">
+                            <ul class="slds-has-dividers_around-space" >
+                              <li class="slds-item">
+                                
+                                <article class="slds-tile slds-tile_board">
+                                  <h3 class="slds-tile__title slds-truncate" title="Journey Name">
```
Comment (Drywkenway) [inline force-app/main/default/lwc/timeline/timeline.html:102]: why is the title journey name?

File: force-app/main/default/lwc/timeline/timeline.html [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+                    
+                        <template if:true={OverdueTasks}>
+
+                      
+                          <div class="task-card">
+                            <ul class="slds-has-dividers_around-space">
+                              <li class="slds-item">
+                                
+                                <article class="slds-tile slds-tile_board">
+                                  <h3 class="slds-tile__title slds-truncate" title="Journey Name">
```
Comment (Drywkenway) [inline force-app/main/default/lwc/timeline/timeline.html:179]: why is the title journey name?

File: force-app/main/default/lwc/timeline/timeline.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        
+        
+        else if (error) {
+            this.error = error;
+            this.records = undefined;
+        }
+
+    }
+
+    handleClick(){
```
Comment (Drywkenway) [inline force-app/main/default/lwc/timeline/timeline.js:52]: what purpose does this method serve?

**Review by Drywkenway — CHANGES_REQUESTED** [review]

File: force-app/main/default/lwc/timeline/timeline.js (hunk 4 above)
Comment (UKJWL) [reply force-app/main/default/lwc/timeline/timeline.js:52]: Was an attempt at something else, deprecated now, so has been deleted.

File: force-app/main/default/lwc/timeline/timeline.html (hunk 1 above)
Comment (UKJWL) [reply force-app/main/default/lwc/timeline/timeline.html:25]: Changed.

File: force-app/main/default/lwc/timeline/timeline.html (hunk 2 above)
Comment (UKJWL) [reply force-app/main/default/lwc/timeline/timeline.html:102]: Copying and pasting the card lead to an oversight, changed.

File: force-app/main/default/lwc/timeline/timeline.html (hunk 3 above)
Comment (UKJWL) [reply force-app/main/default/lwc/timeline/timeline.html:179]: Copying and pasting the card lead to an oversight, changed.

File: force-app/main/default/lwc/onboardingDateUpdate/onboardingDateUpdate.html [hunk 5]
```diff
@@ -0,0 +1,15 @@
+<template>
+    <lightning-quick-action-panel header="Update Onboarding Start Date">
```
[the remark this answers is not in the document]
Comment (UKJWL) [reply force-app/main/default/lwc/onboardingDateUpdate/onboardingDateUpdate.html:2]: So custom label's isn't something I've really heard of. I've gone away and looked into them and I see why you would use them. I have a question. When you import the labels in your .js file, do you import all labels individually, or can you import them all at once?

File: force-app/main/default/lwc/onboardingDateUpdate/onboardingDateUpdate.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+
+export default class onboardingDateUpdate extends LightningElement {
+    @api recordId;
+    objectApiName = "Journey__c";
+
+    handleSuccess(e) {
+        this.dispatchEvent(new CloseActionScreenEvent());
+        this.dispatchEvent(
+            new ShowToastEvent({
```
[the remark this answers is not in the document]
Comment (UKJWL) [reply force-app/main/default/lwc/onboardingDateUpdate/onboardingDateUpdate.js:14]: changed

File: force-app/main/default/lwc/onboardingDateUpdate/onboardingDateUpdate.html [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+        <lightning-record-edit-form record-id={recordId} object-api-name={objectApiName} onsuccess={handleSuccess}>    
+            <lightning-input-field
+                field-name="Onboarding_Start_Date__c">
+            </lightning-input-field>
+            <lightning-button
+                variant="brand"
+                class="slds-m-left_x-small"
+                label="Save"
```
[the remark this answers is not in the document]
Comment (UKJWL) [reply force-app/main/default/lwc/onboardingDateUpdate/onboardingDateUpdate.html:10]: changed

File: force-app/main/default/lwc/timeline/timeline.html [hunk 8]
```diff
@@ -0,0 +1,242 @@
...
+                
+           
+                  <div class = "scroll-container">  
+                      <div class="tasks-container">
+
+
+                   <template for:each={CompletedTasks} for:item="rec">
+                      <div key={rec.fields.Id.value}>
+                    
+                        <template if:true={CompletedTasks}>
```
[the remark this answers is not in the document]
Comment (UKJWL) [reply force-app/main/default/lwc/timeline/timeline.html:17]: Changed

File: force-app/main/default/lwc/timeline/timeline.html [hunk 9]
```diff
@@ -0,0 +1,242 @@
...
+                        <template if:true={OverdueTasks}>
+
+                      
+                          <div class="task-card">
+                            <ul class="slds-has-dividers_around-space">
+                              <li class="slds-item">
+                                
+                                <article class="slds-tile slds-tile_board">
+                                  <h3 class="slds-tile__title slds-truncate" title="Journey Name">
+                                    <p style = "font-weight: 600;">Task Name:</p>
```
[the remark this answers is not in the document]
Comment (UKJWL) [reply force-app/main/default/lwc/timeline/timeline.html:180]: Changed

File: force-app/main/default/lwc/timeline/timeline.html [hunk 10]
```diff
@@ -0,0 +1,242 @@
...
+                        <template if:true={InProgressTasks}>
+
+                      
+                          <div class="task-card">
+                            <ul class="slds-has-dividers_around-space" >
+                              <li class="slds-item">
+                                
+                                <article class="slds-tile slds-tile_board">
+                                  <h3 class="slds-tile__title slds-truncate" title="Journey Name">
+                                    <p style = "font-weight: 600;">Task Name:</p>
```
[the remark this answers is not in the document]
Comment (UKJWL) [reply force-app/main/default/lwc/timeline/timeline.html:103]: changed

File: force-app/main/default/lwc/timeline/timeline.html [hunk 11]
```diff
@@ -0,0 +1,242 @@
...
+        
+      
+                    </div>
+                    </template>
+                      </div> 
+                  </div>
+
+          </lightning-tab>
+
+           <lightning-tab label="Overdue" value="Overdue" title = "Overdue">
```
[the remark this answers is not in the document]
Comment (UKJWL) [reply force-app/main/default/lwc/timeline/timeline.html:161]: changed

File: force-app/main/default/lwc/timeline/timeline.html [hunk 12]
```diff
@@ -0,0 +1,242 @@
+<template>
+    <lightning-card title="Your Journey Progress">
+     <lightning-tabset variant="scoped" active-tab-value = "">
+          
+          <lightning-tab label="Completed" value="Completed" title = "Completed">
```
[the remark this answers is not in the document]
Comment (UKJWL) [reply force-app/main/default/lwc/timeline/timeline.html:5]: changed

File: force-app/main/default/lwc/timeline/timeline.html [hunk 13]
```diff
@@ -0,0 +1,242 @@
...
+        
+      
+                    </div>
+                    </template>
+                      </div> 
+                  </div>
+
+          </lightning-tab>
+
+          <lightning-tab label="In Progress" value="InProgress" title = "In Progress">
```
[the remark this answers is not in the document]
Comment (UKJWL) [reply force-app/main/default/lwc/timeline/timeline.html:84]: changed

File: force-app/main/default/lwc/timeline/timeline.html [hunk 14]
```diff
@@ -0,0 +1,242 @@
...
+                                  <div class="slds-text-heading_small">
+                                    
+                                      <a class="inactiveLink">
+                                          {rec.fields.Name.value}
+                                      </a>
+                                 
+                                  </div>
+
+                               
+                                    <p class="slds-truncate" title="Esimated Completion Date:">Esimated Completion Date:</p>
```
[the remark this answers is not in the document]
Comment (UKJWL) [reply force-app/main/default/lwc/timeline/timeline.html:197]: changed

File: force-app/main/default/lwc/timeline/timeline.html [hunk 15]
```diff
@@ -0,0 +1,242 @@
...
```
[the remark this answers is not in the document]
Comment (UKJWL) [reply force-app/main/default/lwc/timeline/timeline.html:120]: changed

File: force-app/main/default/lwc/timeline/timeline.html [hunk 16]
```diff
@@ -0,0 +1,242 @@
...
```
[the remark this answers is not in the document]
Comment (UKJWL) [reply force-app/main/default/lwc/timeline/timeline.html:43]: changed

File: force-app/main/default/lwc/timeline/timeline.html [hunk 17]
```diff
@@ -0,0 +1,242 @@
...
+                        <template if:true={CompletedTasks}>
+
+                      
+                          <div class="task-card" >
+                            <ul class="slds-has-dividers_around-space"> 
+                              <li class="slds-item">
+                                
+                                <article class="slds-tile slds-tile_board">
+                                  <h3 class="slds-tile__title slds-truncate" title="Journey Name">
+                                    <p style = "font-weight: 600;">Task Name:</p>
```
[the remark this answers is not in the document]
Comment (UKJWL) [reply force-app/main/default/lwc/timeline/timeline.html:26]: changed

File: force-app/main/default/lwc/timeline/timeline.html [hunk 18]
```diff
@@ -0,0 +1,242 @@
+<template>
+    <lightning-card title="Your Journey Progress">
```
[the remark this answers is not in the document]
Comment (UKJWL) [reply force-app/main/default/lwc/timeline/timeline.html:2]: changed

File: force-app/main/default/lwc/timeline/timeline.html [hunk 19]
```diff
@@ -0,0 +1,242 @@
...
+          <lightning-tab label="Completed" value="Completed" title = "Completed">
+
+                  
+                
+           
+                  <div class = "scroll-container">  
+                      <div class="tasks-container">
+
+
+                   <template for:each={CompletedTasks} for:item="rec">
```
[the remark this answers is not in the document]
Comment (UKJWL) [reply force-app/main/default/lwc/timeline/timeline.html:14]: Changed

**Review by Drywkenway — APPROVED** [review]

**Pull request merged**